src/main/java/de/plushnikov/intellij/plugin/extension/LombokMethodReferenceSearcher.java

```diff
@@ -0,0 +1,39 @@
+package de.plushnikov.intellij.plugin.extension;
+
+import com.intellij.openapi.application.QueryExecutorBase;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiReference;
+import com.intellij.psi.search.searches.MethodReferencesSearch;
+import com.intellij.util.Processor;
+import de.plushnikov.intellij.plugin.psi.LombokLightMethodBuilder;
+import org.jetbrains.annotations.NotNull;
+
+public class LombokMethodReferenceSearcher extends QueryExecutorBase<PsiReference, MethodReferencesSearch.SearchParameters> {
+
+  public LombokMethodReferenceSearcher(boolean requireReadAction) {
+    super(requireReadAction);
+    System.out.println("LombokMethodReferenceSearcher(boolean)");
+  }
+
+  public LombokMethodReferenceSearcher() {
+    System.out.println("LombokMethodReferenceSearcher()");
+  }
+
+  @Override
+  public void processQuery(@NotNull MethodReferencesSearch.SearchParameters queryParameters, @NotNull Processor<? super PsiReference> consumer) {
+    PsiMethod queryMethod = queryParameters.getMethod();
+    LombokLightMethodBuilder builder = LombokLightMethodBuilder.getLightMethodBuilder(queryMethod);
+    if (builder != null) {
+      // Each Lombok-generated PsiMethod exists twice.
```

Comment:
Good catch! How do you find it out? I've never thought in that direction until now. Maybe it's also a reason for some other weird problems that have been observed.

Reply:
Hours of looking at the debugger breakpoints inside `SliceUtil` and `ReferencesSearch` and wondering why I sometimes got a LombokLightMethodBuilder, and  sometimes got a `PsiMethodImpl` for the exact same method.

I've tried many ways to get rid of that duplication, but I can't figure out how. I couldn't find a way to properly insert a generated method into the tree(s) of the containing class and still have it reference the correct parent, correct containing file, etc... 

It looks like the only way may be to have our own full hierarchy of `LombokLight*` classes, but that might be a lot of classes. If it works properly, it may totally remove the need for custom search implementations, though.

Also, I couldn't figure out the difference between `LightMethod` and `LightMethodBuilder`, and I'm not sure if makes a difference whether we extend one or the other.

Reply:
LightMethod is more a delegate to some existing PsiMethod.
LightMethodBuilder can be used to create a new (fake) PsiMethod